dist/restore/index.js

```diff
@@ -0,0 +1,143 @@
+
+    // Get all caches available in Bucket for the given cache key
+    const [availableCaches] = await bucket.getFiles({
+      prefix: cacheKey
+    });
+
+    if (typeof availableCaches !== undefined && availableCaches.length > 0) {
+      console.log(`Caches for key ${cacheKey} found in bucket ${bucketName}`);
+    } else {
+      console.log(`No caches found for key ${cacheKey} in the bucket ${bucketName}`);
+    }
+
+    // List all given path with their corresponding cache references (if exists in the bucket)
+    const cacheObjects = pathsString.split('\n').map(cachePath => {
+      const cacheFileName = fileNameForCache(cacheKey, cachePath);
+      const cacheReference = availableCaches.find(cache => {
+        return cache.name == cacheFileName;
+      });
+
+      return [cachePath, cacheReference];
+    });
+
+    const foundCachesMask = cacheObjects.map(cachePath => {
+      return cachePath[1] !== undefined;
+    });
+
+    core.setOutput('cache-hit', foundCachesMask.every(hasCache => hasCache === true));
+
+    // TODO: Check if it is possible to send the paths of the missing caches to the Post Action via environment variable
+    // process.env.NOT_FOUND_CACHES = '[]';
+
+    const filteredCacheObjects = cacheObjects.filter((_, index) => foundCachesMask[index]);
+
+    if (typeof filteredCacheObjects !== undefined && filteredCacheObjects.length > 0) {
+      if (noDownload === 'false') {
+
+        // TODO: Create a download directory
+        // await exec.exec(`mkdir`, [`${downloadDirectory}`]);
+
+        await Promise.all(filteredCacheObjects.map(async (cacheObject) => {
+            await downloadCache(cacheObject[0], cacheObject[1]);
+          }));
+
+        // TODO: Remove upload directory
+          // .then(async () => {
+          //   await exec.exec(`${removeCommand}`, [`${downloadDirectory}`]);
+          // });
+      }
+    }
+  }
+  catch (error) {
+    core.setFailed(error.message);
+  }
+}
+
+function fileNameForCache(key, path) {
+  return [key, '_', md5(path), compressedFileExtension].join('');
+}
+
+async function downloadCache(cachePath, gcsFileObject) {
+  // TODO: Use a download directory
+  const cacheDownloadPath = path.join(
+    // downloadDirectory,
+    gcsFileObject.name);
+
+  gcsFileObject.exists()
+    .then((response) => {
+      const exists = response[0];
+      if (exists) {
+        gcsFileObject.download({
+          destination: cacheDownloadPath
+        })
+        .then(async () => {
+          if (gcsFileObject.name.endsWith(compressedFileExtension)) {
+            // Expand compressed cache and move them to their corresponding paths
+            await exec.exec(`${expandCommand}`, [`${cacheDownloadPath}`, `${cachePath}`], { silent: true });
+          } else {
+            // Move uncompressed caches to their corresponding paths
+            fs.rename(cacheDownloadPath, cachePath, (error) => {
+              if (error) throw error;
+            });
```

Comment:
this is an async call which uses a callback. wrap this with a promise and await it.

```suggestion
            await new Promise((resolve, reject)=>{
                fs.rename(cacheDownloadPath, cachePath, (error) => {
                  if (error) {return reject(error)};
                  resolve()
                });
            }).catch((e)=>{throw e})
```